src/ray/raylet/raylet_client.h

```diff
@@ -0,0 +1,165 @@
+  /// a task to this worker. The caller takes ownership of the returned task
+  /// specification and must free it.
+  ///
+  /// \return The assigned task.
+  ray::raylet::TaskSpecification *GetTask();
+  /// Tell the local scheduler that the client has finished executing a task.
+  ///
+  /// \return ray::Status.
+  ray::Status TaskDone();
+  /// Tell the local scheduler to reconstruct or fetch objects.
+  ///
+  /// \param object_ids The IDs of the objects to reconstruct.
+  /// \param fetch_only Only fetch objects, do not reconstruct them.
+  /// \param current_task_id The task that needs the objects.
+  /// \return int 0 means correct, other numbers mean error.
+  ray::Status FetchOrReconstruct(const std::vector<ObjectID> &object_ids, bool fetch_only,
+                                 const TaskID &current_task_id);
+  /// Notify the local scheduler that this client (worker) is no longer blocked.
+  ///
+  /// \param current_task_id The task that is no longer blocked.
+  /// \return ray::Status.
+  ray::Status NotifyUnblocked(const TaskID &current_task_id);
+  /// Wait for the given objects until timeout expires or num_return objects are
+  /// found.
+  ///
+  /// \param object_ids The objects to wait for.
+  /// \param num_returns The number of objects to wait for.
+  /// \param timeout_milliseconds Duration, in milliseconds, to wait before
+  /// returning.
+  /// \param wait_local Whether to wait for objects to appear on this node.
+  /// \param current_task_id The task that called wait.
+  /// \return A pair with the first element containing the object ids that were
+  /// found, and the second element the objects that were not found.
+  std::pair<std::vector<ObjectID>, std::vector<ObjectID>> Wait(
+      const std::vector<ObjectID> &object_ids, int num_returns,
+      int64_t timeout_milliseconds, bool wait_local, const TaskID &current_task_id);
+  /// Push an error to the relevant driver.
+  ///
+  /// \param The ID of the job that the error is for.
+  /// \param The type of the error.
+  /// \param The error message.
+  /// \param The timestamp of the error.
+  /// \return ray::Status.
+  ray::Status PushError(const JobID &job_id, const std::string &type,
+                        const std::string &error_message, double timestamp);
+  /// Store some profile events in the GCS.
+  ///
+  /// \param profile_events A batch of profiling event information.
+  /// \return ray::Status.
+  ray::Status PushProfileEvents(const ProfileTableDataT &profile_events);
+  /// Free a list of objects from object stores.
+  ///
+  /// \param object_ids A list of ObjectsIDs to be deleted.
+  /// \param local_only Whether keep this request with local object store
+  /// or send it to all the object stores.
+  /// \return ray::Status.
+  ray::Status FreeObjects(const std::vector<ray::ObjectID> &object_ids, bool local_only);
+
+  UniqueID client_id;
+  bool is_worker;
+  JobID driver_id;
+  Language language;
```

Comment:
should we make `client_id`, `is_worker`, and `language` into `const` fields? @pcmoritz